src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/maas/MaasProvider.java

```diff
@@ -0,0 +1,248 @@
+                                                                                   Integer.valueOf(instance.getNumber()));
+
+        // Start deployment(s) by ID or by resources
+        List<Future<Machine>> futureMachines;
+        if (instance.getId() != null) {
+            futureMachines = IntStream.rangeClosed(1, Integer.valueOf(instance.getNumber()))
+                                      .mapToObj(instanceIndexStartAt1 -> maasPollingService.deployMachine(instance.getId(),
+                                                                                                          instance.getInitScript()
+                                                                                                                  .getScripts()[0],
+                                                                                                          tags))
+                                      .collect(Collectors.toList());
+        } else {
+            futureMachines = IntStream.rangeClosed(1, Integer.valueOf(instance.getNumber()))
+                                      .mapToObj(instanceIndexStartAt1 -> maasPollingService.deployMachine(Integer.valueOf(instance.getHardware()
+                                                                                                                                  .getMinCores()),
+                                                                                                          Integer.valueOf(instance.getHardware()
+                                                                                                                                  .getMinRam()),
+                                                                                                          instance.getInitScript()
+                                                                                                                  .getScripts()[0],
+                                                                                                          tags))
+                                      .collect(Collectors.toList());
+        }
+
+        // Retrieve futures (blocking calls)
+        Set<Instance> instances = futureMachines.stream().map(futureMachine -> {
+            try {
+                return futureMachine.get();
+            } catch (InterruptedException | ExecutionException e) {
+                e.printStackTrace();
+                return null;
+            }
+        }).map(this::getInstanceFromMachine).collect(Collectors.toSet());
+
+        // Kill polling timeout tasks
+        maasPollingService.shutdown();
+
+        return instances;
+    }
+
+    @Override
+    public void deleteInstance(Infrastructure infrastructure, String instanceId) {
+        if (!maasProviderClientCache.getMaasClient(infrastructure).releaseMachineById(instanceId)) {
+            throw new RuntimeException("ERROR when deleting MAAS instance : " + instanceId);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00112)